core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -782,6 +785,9 @@ private[storage] class PartiallySerializedBlock[T](
    * `close()` on it to free its resources.
    */
   def valuesIterator: PartiallyUnrolledIterator[T] = {
+    // Close the serialization stream so that the serializer's internal buffers are freed and any
+    // "end-of-stream" markers can be written out so that `unrolled` is a valid serialized stream.
+    serializationStream.close()
```

Comment:
It seems like 'unrolled' may basically be invalid until serializationStream is called.

But it looks like valuesIterator is not the only place where unrolled is used